deployment.md

```diff
@@ -0,0 +1,162 @@
+# ERC20 paymaster deployment
+
+This document describes the deployment process of the ERC20 paymaster and the accompanying infrastructure.
+
+## Prerequisites
+
+### Foundry
+
+Deployments are done using the `foundry` toolchain. Make sure you have it installed.
+
+Read more about installation [on their official documentation](https://book.getfoundry.sh/getting-started/installation).
+
+### Dependencies
+
+ERC20 paymaster uses standalone contracts for its operation:
+
+- 2 Oracles, that provide the price feed for the paymaster
+- Uniswap-v3-compatible liquidity pool
+- ERC20Paymaster itself
+
+#### Oracles
+
+Note that 2 oracles provide the Token <> USD and USD <> ETH price feeds, which require 2 corresponding liquidity pools.
+However, it is possible to operate with only 1 liquidity pool, Token <> ETH. In this case, one oracle will provide the Token <> ETH price, and the other should state the exchange rate of USD <> USD, and therefore can be a fixed oracle.
+
+#### Liqiudity pool
+
+Token <> ETH liquidity pool is required for the operation of the ERC20 paymaster. It is used by the TwapOracle to provide the price feed for the ERC20 paymaster.
+Liqiudity pool deployment is outside the scope of this document.
+
+### Factories
+
+Deployment of the paymaster and its infrastructure happens through the corresponding factory contracts on each chain.
+
+Note that for security purposes, both factories are ownerful, and only the owner can deploy new instances of the paymaster and its infrastructure.
+Therefore, you need to specify the owner address when deploying the factory. You should store the owner address to `$FACTORY_OWNER_ADDRESS` and private key for further usage to `$FACTORY_OWNER_PRIVATE_KEY`.
+You can deploy the factory with any account, so specify its private key in `$DEPLOYER_PRIVATE_KEY`.
+
+To deploy the Oracle factory, run the following command:
+
+```shell
+forge create --optimizer-runs 100 -via-ir src/factory/ERC20OracleOnlyFactory.sol:ERC20OracleOnlyFactory --constructor-args $FACTORY_OWNER_ADDRESS --private-key $DEPLOYER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed factory, that you should save to `$ORACLE_FACTORY_ADDRESS`.
```

Comment:
Consider replacing "save to" with "store in".

"Store in" is more precise when referring to placing a value inside a variable or a memory location. "Save to" can be ambiguous and might suggest saving to an external system, whereas "store in" makes it clear that the value is being assigned to a variable.

```suggestion
The command will return the address of the deployed factory, that you should store in `$ORACLE_FACTORY_ADDRESS`.
```

Reply:
60879dbb5b1bb13b48bb5bc131b6e990d3862a88